src/utils/src/observe-dimensions.ts

```diff
@@ -95,9 +95,10 @@ function getSize(node, entry): Dimensions | null {
  * @returns
  */
 export default function useDimensions<T extends Element>(
+  nodeRef?: RefObject<T>,
   throttleDelay = DEFAULT_THROTTLE_DELAY
 ): [RefObject<T>, Dimensions | null] {
-  const ref = useRef<T>(null);
+  const ref = nodeRef ?? useRef<T>(null);
   const [size, setSize] = useState(null);
 
   useEffect(() => {
```

Comment:
We should probably add `ref?.current` to the dependencies of `useEffect` so that the observers get reattached in case the ref element changes.